### PR TITLE
fix: Add style prop to Bullet component

### LINKED ISSUE #2007: Unable to style Ordered or Unordered lists in Markdown
**Describe the bug**

Unable to style the ordered or unordered list in markdown. Always using the default markdown styles

**Dev environment info (please complete/provide the following information):**

- Package version (stream-chat-react-native/stream-chat-expo) - 5.11.2
- react-native or expo version - React Native (imagine both)
- Device/Emulator (android/ios) and OS version - 16.2, iOS (imagine all tho)

**To Reproduce**

1. Boot the Sample App
2. Update `useStreamChatTheme` to have a custom color for either `listItemBullet` or `listItemNumber`
3. See it still uses the default

**Expected behavior**

Be able to see the custom colour you've provided to `listItemBullet` or `listItemNumber`

**Additional context**

**Screenshots**

### PR DESCRIPTION
**Pull request by aharwood9**
## 🎯 Goal

Style was never added to `Bullet` so was always using the default `defaultMarkdownStyles`. This closes #2007 

## 🛠 Implementation details

- Pass the correct markdown style to the `Bullet` component
- Updated `Bullet` and `ListRow` to use the main style object as that already has all the `defaultMarkdownStyles` and then allows for full customisation. 

## 🎨 UI Changes

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## 🧪 Testing

- Open the Sample App, send a message that contains an ordered or unordered list
- See you have all the default markdown styles
- Update the `useStreamChatTheme` to provide a custom colour to the list
- See the list is now updated with the custom colour

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Documentation is updated
- [ ] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [ ] Expo iOS and Android

### CODE REVIEW

**Comment by aharwood9** [comment]
Happy to revert to the part that enforces that `listRow` and `bullet` always override the custom styles, but felt like this gives us the power to change these if we have a use-case 👍 i.e. change the fontWeight of an ordered list

File: package/src/components/Message/MessageSimple/utils/renderText.tsx [hunk 1]
```diff
@@ -354,13 +360,13 @@ interface BulletProps extends TextProps {
...
 
 const Bullet = ({ index, style }: BulletProps) => (
-  <Text key={0} style={[style, defaultMarkdownStyles.listItemNumber]}>
+  <Text key={0} style={style}>
     {index ? `${index}. ` : '\u2022 '}
   </Text>
 );
 
-const ListRow = (props: PropsWithChildren<ViewProps>) => (
-  <Text style={[props.style, defaultMarkdownStyles.listRow]}>{props.children}</Text>
```
Comment (santhoshvai) [inline package/src/components/Message/MessageSimple/utils/renderText.tsx:363]: same here.. should not be removed in my view

File: package/src/components/Message/MessageSimple/utils/renderText.tsx [hunk 2]
```diff
@@ -354,13 +360,13 @@ interface BulletProps extends TextProps {
 }
 
 const Bullet = ({ index, style }: BulletProps) => (
-  <Text key={0} style={[style, defaultMarkdownStyles.listItemNumber]}>
```
[the remark this answers is not in the document]
Comment (aharwood9) [reply package/src/components/Message/MessageSimple/utils/renderText.tsx:357]: The style object that is passed into this component is constructed here - 

`const styles: MarkdownStyle = {
    ...defaultMarkdownStyles,
    ...markdownStyles...`

So the defaultMarkdownStyles is already applied. So by removing the default it allows for full customisation. As I mention in the comment above, I am happy to revert but then the user will not be able to change `fontWeight` which could be surprising. However, if they don't supply a fontWeight, then the styles object will already have the `defaultMarkdownStyles`

File: package/src/components/Message/MessageSimple/utils/renderText.tsx (hunk 1 above)
Comment (aharwood9) [reply package/src/components/Message/MessageSimple/utils/renderText.tsx:363]: As above 👍

File: package/src/components/Message/MessageSimple/utils/renderText.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (santhoshvai) [reply package/src/components/Message/MessageSimple/utils/renderText.tsx:357]: Thanks did not notice this 👍

**Review by santhoshvai — APPROVED** [review]

**Comment by santhoshvai** [comment]
Thanks for the contribution @aharwood9 🙏

**Pull request merged**